src/ol/source/TileWMS.js

```diff
@@ -104,6 +154,8 @@ class TileWMS extends TileImage {
       transition: options.transition
     });
 
+    this.fixedTileUrlFunction = fixedTileUrlFunction;
```

Comment:
Needs to be
```js
this.fixedTileUrlFunction = fixedTileUrlFunction.bind(this);
```

Reply:
Correct. And it also needs to be run before the `super()` call. What's the reasoning behind the lint rule that prevents `this` before `super`?

Reply:
> What's the reasoning behind the lint rule that prevents this before super?

It's a language requirement, not just a linter rule.

Reply:
In that case, we can't bind any function to `this` before `super`. Therefore, should we check for the existence of `options.tileUrlFunction` later and set it explicitly via `this.setTileUrlFunction()`? The current implementation wants to call `this.fixedTileUrlFunction` when `this.setUrl()` or `this.setUrls()` is called from the super class constructor.

I can also leave it the way it was an add `// @ts-ignore` just before the function.

Reply:
> check for the existence of `options.tileUrlFunction` later and set it explicitly via `this.setTileUrlFunction()`?

That sounds reasonable, yes. Thanks!

Reply:
If we go that route, should we remove the `fixedTileUrlFunction` logic altogether since it isn't usable?

Reply:
Yes, please remove that logic.